File: IMUIKitOC/IMUIKitOCExample/IMUIKitOCExample/Main/NETabbarController.m

// Copyright (c) 2022 NetEase, Inc. All rights reserved.
// Use of this source code is governed by a MIT license that can be
// found in the LICENSE file.

#import "NETabbarController.h"
#import "NENavigationController.h"

#import <NEChatKit/NEChatKit-Swift.h>
#import <NEChatUIKit/NEChatUIKit-Swift.h>
#import <NECommonUIKit/NECommonUIKit-Swift.h>
#import <NEContactUIKit/NEContactUIKit-Swift.h>
#import <NEConversationUIKit/NEConversationUIKit-Swift.h>
#import <NECoreIM2Kit/NECoreIM2Kit-Swift.h>
#import <NECoreKit/NECoreKit-Swift.h>
// #import <NEQChatUIKit/NEQChatUIKit-Swift.h>

@interface NETabbarController ()

@end

@implementation NETabbarController

- (void)viewDidLoad {
  [super viewDidLoad];
  // Do any additional setup after loading the view.
  [self setUpControllers];
}

- (void)setUpControllers {
  //  ChatViewController *chat = [[ChatViewController alloc] init];
  //  NENavigationView *nenav = [[NENavigationView alloc] init];

  [ChatRouter registerFun];

  // 会话列表页
  ConversationController *sessionCtrl = [[ConversationController alloc] init];

  sessionCtrl.view.backgroundColor = [UIColor whiteColor];
  sessionCtrl.tabBarItem = [[UITabBarItem alloc] initWithTitle:NSLocalizedString(@"message", @"")
                                                         image:[UIImage imageNamed:@"chat"]
                                                 selectedImage:[UIImage imageNamed:@"chatSelect"]];
  NENavigationController *sessionNav =
      [[NENavigationController alloc] initWithRootViewController:sessionCtrl];

  // 通讯录
  ContactViewController *contactCtrl = [[ContactViewController alloc] init];
  contactCtrl.tabBarItem =
      [[UITabBarItem alloc] initWithTitle:NSLocalizedString(@"contact", @"")
                                    image:[UIImage imageNamed:@"contact"]
                            selectedImage:[UIImage imageNamed:@"contactSelect"]];
  NENavigationController *contactNav =
      [[NENavigationController alloc] initWithRootViewController:contactCtrl];

  // 圈组

  //  QChatHomeViewController *qchatCtrl = [[QChatHomeViewController alloc] init];
  //  qchatCtrl.tabBarItem =
  //      [[UITabBarItem alloc] initWithTitle:NSLocalizedString(@"qchat", @"")
  //                                    image:[UIImage imageNamed:@"qchat_tabbar_icon"]
  //                            selectedImage:[UIImage imageNamed:@"qchat_tabbar_icon"]];
  //  NENavigationController *qchatNav =
  //      [[NENavigationController alloc] initWithRootViewController:qchatCtrl];

  self.tabBar.backgroundColor = [UIColor whiteColor];
  self.viewControllers = @[ sessionNav, contactNav ];
  self.selectedIndex = 0;
}

@end

File: app/NETabBarController.swift

// Copyright (c) 2022 NetEase, Inc.  All rights reserved.
// Use of this source code is governed by a MIT license that can be found in the LICENSE file.

import UIKit
import NIMSDK
import NEKitCore
import NEKitQChatUI
import NEKitCoreIM
import NEKitConversationUI
import NEKitTeamUI
import NEKitChatUI
import NEKitContactUI

class NETabBarController: UITabBarController {

    private var sessionUnreadCount = 0
    private var contactUnreadCount = 0

    
    override func viewDidLoad() {
        super.viewDidLoad()
        setUpControllers()
        setUpSessionBadgeValue()
        setUpContactBadgeValue()
        NIMSDK.shared().conversationManager.add(self)
        NIMSDK.shared().systemNotificationManager.add(self)
    }

    func setUpControllers(){
        //chat
        let chat = ConversationController()
        chat.view.backgroundColor = UIColor.init(hexString: "#e9eff5")
        chat.tabBarItem = UITabBarItem(title: "消息", image: UIImage(named: "chat"), selectedImage: UIImage(named: "chatSelect")?.withRenderingMode(.alwaysOriginal))
        let chatNav = NENavigationController(rootViewController: chat)
        
        //qchat
        let qchat = QChatHomeViewController()
        qchat.view.backgroundColor = UIColor.init(hexString: "#e9eff5")
        qchat.tabBarItem = UITabBarItem(title: "圈组", image: UIImage(named: "qchat_tabbar_icon"), selectedImage: UIImage(named: "qchat_tabbar_icon")?.withRenderingMode(.alwaysOriginal))
        let qChatNav = NENavigationController(rootViewController: qchat)
        
        
        // Contacts
        let contactVC = ContactsViewController()
        contactVC.tabBarItem = UITabBarItem(title: "通讯录", image: UIImage(named: "contact"), selectedImage: UIImage(named: "contactSelect")?.withRenderingMode(.alwaysOriginal))
        contactVC.title = "通讯录"
        let contactsNav = NENavigationController(rootViewController: contactVC)
        
        // Me
        let meVC = MeViewController()
        meVC.view.backgroundColor = UIColor.white
        meVC.tabBarItem = UITabBarItem(title: "我", image: UIImage(named: "person"), selectedImage: UIImage(named: "personSelect")?.withRenderingMode(.alwaysOriginal))
        let meNav = NENavigationController(rootViewController: meVC)

        tabBar.backgroundColor = .white
        self.viewControllers = [chatNav,qChatNav,contactsNav,meNav]
        self.selectedIndex = 0
    }
    
    func setUpSessionBadgeValue(){
        sessionUnreadCount = ConversationRepo().getMsgUnreadCount(notify: true)
        if sessionUnreadCount > 0  {
            self.tabBar.showBadgOn(index: 0)
        }else {
            self.tabBar.hideBadg(on: 0)
        }
    }
    
    func setUpContactBadgeValue() {
        contactUnreadCount = NIMSDK.shared().systemNotificationManager.allUnreadCount()
        if contactUnreadCount > 0 {
            self.tabBar.showBadgOn(index: 2)
        }else {
            self.tabBar.hideBadg(on: 2)
        }
    }
    
    private func refreshSessionBadge(){
        setUpSessionBadgeValue()
    }
    
    deinit {
        NIMSDK.shared().systemNotificationManager.remove(self)
        NIMSDK.shared().conversationManager.remove(self)
    }
}

extension NETabBarController:NIMConversationManagerDelegate {
    
    func didAdd(_ recentSession: NIMRecentSession, totalUnreadCount: Int) {
        refreshSessionBadge()
    }
    
    func didUpdate(_ recentSession: NIMRecentSession, totalUnreadCount: Int) {
        refreshSessionBadge()
    }
    
    func didRemove(_ recentSession: NIMRecentSession, totalUnreadCount: Int) {
        refreshSessionBadge()
    }
}

extension NETabBarController:NIMSystemNotificationManagerDelegate {
    func onSystemNotificationCountChanged(_ unreadCount: Int) {
        contactUnreadCount = unreadCount
        setUpContactBadgeValue()
    }
}
